### PR TITLE
Fix small repeated issues

### PR DESCRIPTION
**Pull request by visuve**
Some static analysis would do good for the repo.

### CODE REVIEW

**Comment by busterb** [comment]
The ++i / i++ do not make any difference, I believe the compiler will optimize out the post-increment regardless if there is nothing reading the value. But agree on the leaks, thanks!

File: c/meterpreter/source/server/win/metsrv_test.c [hunk 1]
```diff
@@ -76,5 +76,10 @@ int main(int argc, char **argv)
 
 	} while (0);
 
+	if (dllPath)
```
Comment (busterb) [inline c/meterpreter/source/server/win/metsrv_test.c]: no need for the check here, just freeing is fine

Comment (visuve) [reply c/meterpreter/source/server/win/metsrv_test.c]: Fixed

**Comment by visuve** [comment]
I think people tend to believe more on what the compilers can optimize than they actually do. 

However I tested your belief and it seems correct. In Visual Studio 2017 ++i/i++ in release builds generates exactly same assembly. However in debug, there is a slight disadvantage over i++.

IMO if a thing can even in theory improve performance and not reduce readability, it's worth doing :-)

**Comment by busterb** [comment]
Not sold on pre-increment being any more readable than post, and in the C++ this tweaked, ++ could do literally anything at all. But thanks all the same!

**Pull request merged**